internal/c/parts/audio/audio.cpp

```diff
@@ -1694,64 +1565,73 @@ static ma_result InitializeSoundFromMemory(const void *buffer, size_t size, int3
+/// @param qbsRequirements This is leftover from the old QB64-SDL days. But we use this to pass some parameters like 'stream'
+/// @param passed Optional parameter flag
+/// @return Returns a valid sound handle (> 0) if successful or 0 if it fails
+int32_t func__sndopen(qbs *qbsFileName, qbs *qbsRequirements, int32_t passed) {
+    if (!audioEngine.isInitialized || !qbsFileName->len)
         return INVALID_SOUND_HANDLE;
 
-    if (!fileNameZ)
-        fileNameZ = qbs_new(0, 0);
-
-    if (!reqs)
-        reqs = qbs_new(0, 0);
-
     // Allocate a sound handle
     int32_t handle = audioEngine.CreateHandle();
-    if (handle < 1) // We are not expected to open files with handle 0
+    if (handle < 1)
         return INVALID_SOUND_HANDLE;
 
-    // Set some handle properties
-    audioEngine.soundHandles[handle]->type = SoundHandle::Type::STATIC;
+    // Set some sound default properties
+    audioEngine.soundHandles[handle]->type = SoundHandle::Type::STATIC; // set the sound type to static by default
+    audioEngine.soundHandles[handle]->maFlags |= MA_SOUND_FLAG_DECODE;  // set the sound to decode completely first before playing (QB64 default)
+    audioEngine.soundHandles[handle]->maFlags |= MA_SOUND_FLAG_ASYNC;   // set the sound to decode asynchronously by default
+    bool fromMemory = false;                                            // we'll assume we are loading the sound from disk
 
-    // Prepare the requirements string
-    if (passed && requirements->len)
-        qbs_set(reqs, qbs_ucase(requirements)); // Convert tmp str to perm str
+    AUDIO_DEBUG_PRINT("Sound set to fully decode asynchronously");
 
-    // Set the flags to specify how we want the audio file to be opened
-    if (passed && requirements->len && func_instr(1, reqs, qbs_new_txt(REQUIREMENT_STRING_STREAM), 1)) {
-        audioEngine.soundHandles[handle]->maFlags |= MA_SOUND_FLAG_STREAM; // Check if the user wants to stream the file
-        AUDIO_DEBUG_PRINT("Sound will stream");
-    } else {
-        audioEngine.soundHandles[handle]->maFlags |= MA_SOUND_FLAG_DECODE; // Else decode and load the whole sound in memory
-        AUDIO_DEBUG_PRINT("Sound will be fully decoded");
+    if (passed && qbsRequirements->len) {
+        // Parse all flags in one go
+        std::string requirements(reinterpret_cast<char const *>(qbsRequirements->chr), qbsRequirements->len);
+        std::transform(requirements.begin(), requirements.end(), requirements.begin(), ::tolower);
+
+        // Check if user wants to set the stream flag
+        if (requirements.find("stream") != std::string::npos) {
+            audioEngine.soundHandles[handle]->maFlags |= MA_SOUND_FLAG_STREAM;
+            AUDIO_DEBUG_PRINT("Sound will be streamed");
+        }
+
+        // Check if the user wants to unset the decode flag
+        if (requirements.find("nodecode") != std::string::npos) {
```

Comment:
So, we were setting up those that way previously. But it turns out, it's really not required.

Without `nodecode` the sound will be decoded "on-load" to PCM format else it will be loaded in the original compressed format and decoded "on-mix".
Without `steam` the entire file is loaded into memory (either decoded as PCM or in the original format based on the `nodecode` flag). However, with `steam` only small chunks are either decoded or loaded in the original format.

So, in the end, it boils down to when the decoding happens - on load or during mixing. I ran some tests to confirm the behavior.

I think doing it this way allows the user a lot of control. The only problem is _MEMSOUND. But for that, we do check if "stream" is off and "decode" enabled.